docs/cluster_and_device_type_dev/how_to_add_new_dts_and_clusters.md

```diff
@@ -0,0 +1,248 @@
+       official YAML representing the official test plan.
+4. Add the test plan, using the templates below:
+
+    1. [cluster_test_plan_template.adoc](https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/template/cluster_test_plan_template.adoc)
+    2. [test_plan_template.adoc](https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/template/test_plan_template.adoc)
+
+    Also, join this private Slack channel: csg-tt-test-plans
+
+5. Note: the CHIP-Tool reference client is generated from XML
+6. If applicable, add tests:
+    1. For relatively simple tests, add YAML tests here:
+       [src/app/tests/suites/certification](https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/certification).
+       Remember to update this file:
+       [src/app/tests/suites/certification/PICS.yaml](https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/certification/PICS.yaml)
+    2. For more complex tests, add Python tests here:
+       [src/python_testing](https://github.com/project-chip/connectedhomeip/tree/master/src/python_testing)
+    3. To add tests to CI, if applicable:
+        1. Add the Python tests here:
+           [.github/workflows/tests.yaml](https://github.com/project-chip/connectedhomeip/tree/master/.github/workflows/tests.yaml).
+           Remember to provide all arguments required for each Python script,
+           such as the endpoint PIXIT.
+        2. Add the YAML tests by editing this file:
+           [src/app/tests/suites/ciTests.json](https://github.com/project-chip/connectedhomeip/tree/master/src/app/tests/suites/ciTests.json)
+            1. Create a section ("MyDevice") which lists all YAML tests for your
+               device
+            2. Add the section's name to the list of items named "collection"
+            3. Do a ZAP code regen: `./scripts/tools/zap_regen_all.py`.
+7. Add the device type spec to the test plan tools:
+
+    1. [tools/device_type_requirements](https://github.com/CHIP-Specifications/chip-test-plans/tree/master/tools/device_type_requirements)
+    2. The file above is used by
+       [src/app/tests/suites/certification/Test_TC_DESC_2_1.yaml](https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/certification/Test_TC_DESC_2_1.yaml)
+
+    Note: the plan is to make the DM tools generate the device type requirements
+    data based on the spec, so the above will become obsolete.
+
+8. Add the device type to Chef:
+   [examples/chef/devices](https://github.com/project-chip/connectedhomeip/tree/master/examples/chef/devices)
+
+## Q & A
+
+**Q1**: What kind of devices can be used for the test events? Can one of them be
+the example cluster server app running on a RasPI? Do the independent
+realizations need to run on vendor hardware or can they also run on generic
+hardware, such as ESP32 or RasPI?
+
+**A1**: one realization can be the test harness + the all clusters example app +
```

Comment:
Suggest just giving a link to the cert policy or something here. They have a whole doc that deals with test events. Granted, it's not exactly easy to read. But that is the source of truth, and I don't want to get out of sync with that.

Reply:
Good point.  Is the cert policy public?

Reply:
Per sidebar convo, cert policy is not public, so not going to put in a link.